src/pipecat/transports/services/pipecat_cloud.py

```diff
@@ -0,0 +1,313 @@
+from pipecat.transports.network.fastapi_websocket import (
+    FastAPIWebsocketParams,
+    FastAPIWebsocketTransport,
+    FrameSerializer,
+)
+from pipecat.transports.network.small_webrtc import (
+    SmallWebRTCConnection,
+    SmallWebRTCTransport,
+)
+from pipecat.transports.services.daily import (
+    DailyDialinSettings,
+    DailyParams,
+    DailyTranscriptionSettings,
+    DailyTransport,
+)
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class WebRTCSessionArguments(SessionArguments):
+    """WebRTC based agent session arguments. The arguments are received by the
+    bot() entry point.
+    """
+
+    webrtc_connection: SmallWebRTCConnection
+
+
+class PipecatCloudParams(TransportParams):
+    """Parameters for PipecatCloudTransport.
+
+    This class combines parameters from all transport types:
+    - TransportParams (inherited)
+    - FastAPIWebsocketParams
+    - DailyParams
+    """
+
+    # FastAPIWebsocketParams fields
+    add_wav_header: bool = False
+    serializer: Optional[FrameSerializer] = None
+    session_timeout: Optional[int] = None
+
+    # DailyParams fields
+    api_url: str = "https://api.daily.co/v1"
+    api_key: str = ""
+    dialin_settings: Optional[DailyDialinSettings] = None
+    transcription_enabled: bool = False
+    transcription_settings: DailyTranscriptionSettings = DailyTranscriptionSettings()
+
+    def to_transport_params(self) -> TransportParams:
+        """Convert to TransportParams."""
+        return self
+
+    def to_websocket_params(self) -> FastAPIWebsocketParams:
+        """Convert to FastAPIWebsocketParams."""
+        base_params = self.model_dump()
+        # Remove WebSocket-specific fields since we'll add them explicitly
+        websocket_fields = ["add_wav_header", "serializer", "session_timeout"]
+        for field in websocket_fields:
+            base_params.pop(field, None)
+
+        return FastAPIWebsocketParams(
+            **base_params,
+            add_wav_header=self.add_wav_header,
+            serializer=self.serializer,
+            session_timeout=self.session_timeout,
+        )
+
+    def to_daily_params(self) -> DailyParams:
+        """Convert to DailyParams."""
+        base_params = self.model_dump()
+        # Remove Daily-specific fields since we'll add them explicitly
+        daily_fields = [
+            "api_url",
+            "api_key",
+            "dialin_settings",
+            "transcription_enabled",
+            "transcription_settings",
+        ]
+        for field in daily_fields:
+            base_params.pop(field, None)
+
+        return DailyParams(
+            **base_params,
+            api_url=self.api_url,
+            api_key=self.api_key,
+            dialin_settings=self.dialin_settings,
+            transcription_enabled=self.transcription_enabled,
+            transcription_settings=self.transcription_settings,
+        )
+
+
+class PipecatCloudSessionArguments:
```

Comment:
I don't think we really need this class. In the example we already have a way to differentiate between `local_daily`, `bot`, `local_webrtc`. We can just use the right SessionArguments there. For example:

```
async def local_webrtc(webrtc_connection):
    await main(WebRTCSessionArguments(webrtc_connection=webrtc_connection))
```

And we will still be able to do `isinstance(args,...)`.

Reply:
If we were to add more fields to each SessionArguments the `__init__` parameters would not be really nice.